artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -287,6 +288,9 @@ public void bufferReceived(Object connectionID, ActiveMQBuffer buffer) {
          if (logger.isTraceEnabled()) {
             traceBufferReceived(connectionID, command);
          }
+         if (AuditLogger.isAnyLoggingEnabled()) {
+            AuditLogger.setRemoteAddress(getRemoteAddress());
```

Comment:
this won't work...

openWireActor.act() will transfer the execution to a different thread.

you will lose the ThreadLocal here.

The netty loop is reused for many connections.

Reply:
this could eventually fail in other protocols as well. you would endup informing a wrong remote address for Audit Operations.

Reply:
ok, I will address

Reply:
Just an idea. We have OperationContext, which will hold operations to the Session, and it has a thread local already.

Why don't you somehow inject the session on the operation context and use the thread local from there. We recover it back whenever we do a new operation on the session. It is a reliable place.

Reply:
I mean.. why don't you inject the data you need on the operation context.